File: Linguaggi/Simulazioni/40/Python/Cliente.py

from Noleggio import Noleggio

class Cliente:
    
    def __init__(self, nomeCognome):
        self.nomeCognome = nomeCognome
        self.noleggi = []
        
    def addNoleggio(self, i):
        self.noleggi.append(i)
        
    def __str__(self):
        return self.nomeCognome + " " + str(len(self.noleggi)) + " ".join([str(v) for v in self.noleggi])

    
    

File: Linguaggi/Simulazioni/40/Java/Cliente.java
import java.util.*;

public class Cliente {
    protected String nomeCognome;
    protected List<Noleggio> noleggi;

    public Cliente(String nomeCognome) {
        this.nomeCognome = nomeCognome;
        this.noleggi = new LinkedList<Noleggio>();
    }

    public void addNoleggio(String targa, int giorni, double costo) {
        this.noleggi.add(new Noleggio(targa, giorni, costo));
    }

    public double getCostoTotale() {
        double costoTotale = 0;
        for(Noleggio noleggio : this.noleggi) {
            costoTotale += noleggio.getCostoNoleggio();
        }

        return costoTotale;
    }

    public String toString() {
        return this.nomeCognome + " "+ this.getCostoTotale() + " " + noleggi.size() + " " + noleggi;
    }
}
